reconcile/utils/gitlab_api.py

```diff
@@ -130,7 +130,7 @@ def __init__(
     @cached_property
     def project_main_branch(self) -> str:
         return next(
-            (b.name for b in self.project.branches.list() if b.default),
+            (b.name for b in self.project.branches.list(get_all=True) if b.default),
```

Comment:
use `iterator=True` may save some requests if default branch returns in early pages

Reply:
Applied in [0bc805f](https://github.com/app-sre/qontract-reconcile/pull/4499/commits/0bc805ff02dd78b3980ac90cc8418dd0652bf972)